api/envoy/api/v2/core/base.proto

```diff
@@ -233,14 +233,17 @@ message RuntimeFeatureFlag {
 // Header name/value pair.
 message HeaderValue {
   // Header name.
-  string key = 1 [(validate.rules).string = {min_bytes: 1 max_bytes: 16384}];
+  string key = 1 [
+    (validate.rules).string = {min_bytes: 1 max_bytes: 16384 well_known_regex: HTTP_HEADER_NAME}
```

Comment:
Changes to this file may result in the rejection of configs that were previously considered valid.  Is there a way to disable validation for some subset of the fields?

Reply:
I think fundamentally, @envoyproxy/api-shepherds need to decide if this is a breaking change. Arguably the RFC header name/value compliance was an implicit assumption already in the API and Envoy. Taking the opposite argument, there may be folks who relied on this previously unenforced behavior that will be affected. 

My $0.02 is that some aspects of this, e.g. eliminated \r\n\0, were already implied contract and this does not break for those. For other forbidden chars, I think this is a breaking change.

Reply:
I think then that the right thing might be to replace the ones that will be checked for \r\n\0 with boilerplate custom regex for \r\n\0 with a deprecation and replacement field with the full validations... 
for the fields that did the full `headerIsValid` http utility checks, which match the full forbidden char set, those can probably stay